cost-analyzer/templates/_helpers.tpl

```diff
@@ -157,6 +157,26 @@ will result in failure. Users are asked to select one of the two presently-avail
   {{- end -}}
 {{- end -}}
 
+{{/*
+Cluster Controller API Key check
+*/}}
+{{- define "clusterController.kubecostAPIKeyCheck" -}}
+  {{- if .Values.clusterController.enabled }}
+    {{- if and .Values.clusterController.kubecostAPIKey ((eq .Values.clusterController.createClusterControllerSecret false)) }}
+      {{- fail "\n\nConfiguration error:\nWhen setting clusterController.kubecostAPIKey, clusterController.createClusterControllerSecret must be true." -}}
+    {{- end -}}
+    {{- if and (not .Values.clusterController.createClusterControllerSecret) .Values.clusterController.secretName }}
```

Comment:
One other concern with this check, is that it expects a certain value inside `.Values.clusterController.secretName`. I'm good to approve for now, but should we create a Jira to track/resolve the conflicting usage of `.Values.clusterController.secretName` (details in a different PR comment).